memory-external/hacks/hack.hpp

```diff
@@ -76,89 +48,27 @@ namespace hack {
-			if (config::team_esp && (playerTeam == localTeam))
-				continue;
-
-			const std::uint32_t playerPawn = process->read<std::uint32_t>(player + updater::offsets::m_hPlayerPawn);
-
-			const uintptr_t list_entry2 = process->read<uintptr_t>(entity_list + 0x8 * ((playerPawn & 0x7FFF) >> 9) + 16);
-			if (!list_entry2)
-				continue;
-
-			const uintptr_t pCSPlayerPawn = process->read<uintptr_t>(list_entry2 + 120 * (playerPawn & 0x1FF));
-			if (!pCSPlayerPawn)
-				continue;
-
-			const int playerHealth = process->read<int>(pCSPlayerPawn + updater::offsets::m_iHealth);
-			const int playerArmor = process->read<int>(pCSPlayerPawn + updater::offsets::m_ArmorValue);
-			if (playerHealth <= 0 || playerHealth > 100)
-				continue;
-
-			if (config::team_esp && (pCSPlayerPawn == localPlayer))
-				continue;
-
-			std::string playerName = "Invalid Name";
-			const DWORD64 playerNameAddress = process->read<DWORD64>(player + updater::offsets::m_sSanitizedPlayerName);
-			if (playerNameAddress) {
-				char buf[256];
-				process->read_raw(playerNameAddress, buf, sizeof(buf));
-				playerName = std::string(buf);
-			}
-
-			const Vector3 origin = process->read<Vector3>(pCSPlayerPawn + updater::offsets::m_vecOrigin);
-			const Vector3 head = { origin.x, origin.y, origin.z + 75.f };
-
-			if (origin.x == localOrigin.x && origin.y == localOrigin.y && origin.z == localOrigin.z)
-				continue;
-
-			
-
-			if (config::render_distance != -1 && (localOrigin - origin).length2d() > config::render_distance)
-				continue;
-
-			if (origin.x == 0 && origin.y == 0)
-				continue;
-
-			const Vector3 screenPos = origin.world_to_screen(view_matrix);
-			const Vector3 screenHead = head.world_to_screen(view_matrix);
-			
+		for (auto player = g_game.players.begin(); player < g_game.players.end(); player++) {
+			const Vector3 screenPos = g_game.world_to_screen(&player->origin);
+			const Vector3 screenHead = g_game.world_to_screen(&player->head);
 
 			if (screenPos.z >= 0.01f) {
 				const float height = screenPos.y - screenHead.y;
 				const float width = height / 2.4f;
 
-				float distance = localOrigin.calculate_distance(origin);
+				float distance = g_game.localOrigin.calculate_distance(player->origin);
 				int roundedDistance = std::round(distance / 10.f);
 
-				if (config::show_head_tracker)
-				{
-					const uintptr_t gamescene = process->read<uint64_t>(pCSPlayerPawn + 0x310);
-					const uintptr_t bonearray = process->read<uint64_t>(gamescene + 0x160 + 0x80);
-					const Vector3 headPos = process->read<Vector3>(bonearray + 6 * 32);
-					const Vector3 screenHeadPos = headPos.world_to_screen(view_matrix);
+				if (config::show_head_tracker) {
+					const Vector3 screenHeadPos = g_game.world_to_screen(&player->skull);
+					const float head_height = (screenHeadPos.y - screenHeadPos.y) / 8;
+					const float head_width = (head_height / 2.4f) / 4;
 
 					render::DrawCircle(
 						g::hdcBuffer,
 						screenHeadPos.x,
 						screenHeadPos.y,
-						width / 5,
```

Comment:
This needs to stay as width / 5 for head tracker to be distance based, thanks :D